### PR TITLE
DATAJPA-931 - Avoid unnecessary merging on save.

### PR DESCRIPTION
**Pull request by schauder**
Checking if entity is already attached to entity manager before calling merge.
    
Fixed one test that was relying on the implicit flush triggered by the save.

See also: https://vladmihalcea.com/2016/07/19/jpa-persist-and-merge/

### CODE REVIEW

File: src/test/java/org/springframework/data/jpa/domain/support/AuditingEntityListenerTests.java [hunk 1]
```diff
@@ -89,7 +90,7 @@ public void auditsTransitiveEntitiesCorrectly() {
 		role.setName("ADMIN");
 
 		user.addRole(role);
-		repository.save(user);
```
Comment (olivergierke) [inline src/test/java/org/springframework/data/jpa/domain/support/AuditingEntityListenerTests.java]: Why is this change needed? And if applied, shouldn't it be `saveAndFlush(…)`?

**Comment by schauder** [comment]
merged: https://github.com/spring-projects/spring-data-jpa/commit/26f70bf1f002ac34f47a42bf06268f754ef0df6c
